### PR TITLE
Handle HTTP specific errors then other errors when downloading rustc

### LINKED ISSUE #10882: Treat 404 responses when downloading rustc differently from network errors
```
16:03 <fridsun> $ ./mach build --release
16:03 <fridsun> looking for rustc at /c/Users/yang/Documents/workspace-rust/servo/.servo/rust/2016-04-24/rustc-nightly-x86_64-unknown/rustc/bin/rustc
16:03 <fridsun> Downloading Rust compiler...
16:03 <fridsun> Error downloading Rust compiler; are you connected to the internet?
```

### PR DESCRIPTION
**Pull request by askeing**
fix #10882

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10901)

### CODE REVIEW

**Comment by Wafflespeanut** [comment]
Not entirely sure about what we should do for unsupported platforms. [The issue seemed to address windows](https://github.com/servo/servo/issues/10882#issue-151478374) - maybe if there's a way to get the build for windows? I'll let someone else look into this...

**Comment by jdm** [comment]
We should not be trying to compensate for misconfigured environments. The person in question was using the wrong build shell, so subsequent steps would have failed.

File: python/servo/bootstrap_commands.py [hunk 1]
```diff
@@ -63,11 +63,13 @@ def download(desc, src, writer, start_byte=0):
 
         if not dumb:
             print()
-    except urllib2.URLError:
-        print("Error downloading Rust compiler; are you connected to the internet?")
-        sys.exit(1)
     except urllib2.HTTPError, e:
         print("Download failed (%d): %s - %s" % (e.code, e.reason, src))
+        if e.code == 403:
+            print("Platform support information: https://doc.rust-lang.org/book/getting-started.html#platform-support")
```
Comment (jdm) [inline python/servo/bootstrap_commands.py]: I think "No Rust compiler binary available for this platform. Please see https://github.com/servo/servo/#prerequisites" would be more useful.

Comment (askeing) [inline python/servo/bootstrap_commands.py]: you are right, i'll modify it :)

**Comment by askeing** [comment]
`test-tidy` failed due to: `./python/servo/bootstrap_commands.py:69: Line is longer than 120 characters`.
need to merge again.

**Comment by Wafflespeanut** [comment]
@askeing Squash the commits please...

**Comment by askeing** [comment]
@Wafflespeanut sorry for forgetting to rebase commits, thx for your reminder :)

**Comment by Wafflespeanut** [comment]
Hmm, highfive should've unassigned nox and assigned jdm here... (meh)

**Pull request merged**